tests/unit/test_utils.py

```diff
@@ -1356,6 +1363,39 @@ def test_redact_password_from_url(auth_url, expected_url):
     assert url == expected_url
 
 
+class TestHiddenText:
+
+    def test_default_redacted(self):
+        hidden = HiddenText('my-secret')
+        assert repr(hidden) == "<HiddenText '****'>"
+        assert str(hidden) == '****'
+        assert hidden.redacted == '****'
+        assert hidden.secret == 'my-secret'
+
+    def test_custom_redacted(self):
+        hidden = HiddenText('my-secret', redacted='######')
+        assert repr(hidden) == "<HiddenText '######'>"
+        assert str(hidden) == '######'
+        assert hidden.redacted == '######'
+        assert hidden.secret == 'my-secret'
+
+
+def test_hide_value():
+    hidden = hide_value('my-secret')
+    assert repr(hidden) == "<HiddenText '****'>"
+    assert str(hidden) == '****'
+    assert hidden.redacted == '****'
+    assert hidden.secret == 'my-secret'
+
+
+def test_hide_url():
+    hidden_url = hide_url('https://user:password@example.com')
+    assert repr(hidden_url) == "<HiddenText 'https://user:****@example.com'>"
```

Comment:
We should consider and test for single-token auth. IMO if only 1 part is provided then it should be redacted.

Reply:
That sounds reasonable. Except it's orthogonal to this PR IMO as it's a change to `redact_password_from_url()` (which `hide_url()` delegates to). Want to file that issue / PR?

Reply:
Done: #6891. Might be considered a good first issue? It should be a 2 line change and 2 additional lines for tests. If you think so then I can add some more details to the issue.

Reply:
Sounds good. 👍